engine/src/main/scala/cromwell/engine/workflow/workflowstore/WorkflowStoreEngineActor.scala

```diff
@@ -0,0 +1,166 @@
+package cromwell.engine.workflow.workflowstore
+
+import akka.actor.{ActorLogging, ActorRef, LoggingFSM, Props}
+import cats.data.NonEmptyList
+import cromwell.core.Dispatcher._
+import cromwell.engine.workflow.WorkflowManagerActor
+import cromwell.engine.workflow.WorkflowManagerActor.WorkflowNotFoundException
+import cromwell.engine.workflow.workflowstore.WorkflowStoreEngineActor.WorkflowStoreActorState
+import cromwell.engine.workflow.workflowstore.WorkflowStoreState.StartableState
+import WorkflowStoreEngineActor._
+import cromwell.core.WorkflowId
+import cromwell.engine.workflow.workflowstore.WorkflowStoreActor._
+import cromwell.services.metadata.MetadataService.MetadataPutAcknowledgement
+import org.apache.commons.lang3.exception.ExceptionUtils
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+final case class WorkflowStoreEngineActor(store: WorkflowStore, serviceRegistryActor: ActorRef)
+  extends LoggingFSM[WorkflowStoreActorState, WorkflowStoreActorData] with ActorLogging {
+
+  implicit val ec: ExecutionContext = context.dispatcher
+
+  startWith(Unstarted, WorkflowStoreActorData(None, List.empty))
+  self ! InitializerCommand
+
+  when(Unstarted) {
+    case Event(InitializerCommand, _) =>
+      val work = store.initialize map { _ =>
+        log.debug("Workflow store initialization successful")
+      }
+      addWorkCompletionHooks(InitializerCommand, work)
+      goto(Working) using stateData.withCurrentCommand(InitializerCommand, sender)
+    case Event(x: WorkflowStoreActorEngineCommand, _) =>
+      stay using stateData.withPendingCommand(x, sender)
+  }
+
+  when(Idle) {
+    case Event(cmd: WorkflowStoreActorEngineCommand, _) =>
+      if (stateData.currentOperation.nonEmpty || stateData.pendingOperations.nonEmpty) {
+        log.error("Non-empty WorkflowStoreActorData when in Idle state: {}", stateData)
+      }
+      startNewWork(cmd, sender, stateData.withCurrentCommand(cmd, sender))
+  }
+
+  when(Working) {
+    case Event(WorkDone, data) =>
+      val newData = data.pop
+      newData.currentOperation match {
+        case None => goto(Idle) using newData
+        case Some(WorkflowStoreActorCommandWithSender(cmd, sndr)) => startNewWork(cmd, sndr, newData)
+      }
+    case Event(cmd: WorkflowStoreActorEngineCommand, data) => stay using data.withPendingCommand(cmd, sender)
+  }
+
+  whenUnhandled {
+    case Event(MetadataPutAcknowledgement(_), _) =>
```

Comment:
covered by the now poorly named #1811 - I had a comment somewhere in here (perhaps now gone) that this PR effectively removes these but doesn't remove stuff like this